go/host/enclave/service.go

```diff
@@ -200,27 +175,108 @@ func (e *Service) Unsubscribe(id rpc.ID) error {
 	return e.GetEnclaveClient().Unsubscribe(id)
 }
 
-// promoteNewActiveSequencer is a background goroutine that promotes a new active sequencer at startup or when the current one fails.
-// It will never give up, it just cycles through current enclaves until one can be successfully promoted.
-func (e *Service) promoteNewActiveSequencer() {
-	for e.activeSequencerID.Load() == _noActiveSequencer && e.running.Load() {
-		for _, guardian := range e.enclaveGuardians {
-			enclID := guardian.GetEnclaveID()
-			e.logger.Info("Attempting to promote new sequencer.", log.EnclaveIDKey, enclID)
-			err := guardian.PromoteToActiveSequencer()
+// managePeriodicBatches is a background goroutine that triggers periodic batch production and ensures there is always
+// exactly one active sequencer enclave for that.
+// It will promote a new active sequencer if the current one fails or is unavailable.
+func (e *Service) managePeriodicBatches() {
+	interval := e.batchInterval
+	if interval == 0 {
+		interval = 1 * time.Second
+	}
+	// we use ticks rathen than sleeps to maintain batch production cadence
+	batchProductionTicker := time.NewTicker(interval)
+
+	for e.running.Load() {
+		// wait for the next tick
+		<-batchProductionTicker.C
```

Comment:
I would add a ``switch`` here even if it does nothing, just to signal at a glance that this is one of those "wait on channels" loop

Reply:
Don't worry about that too much. Only if it's easy